tajo-core/src/main/java/org/apache/tajo/querymaster/Query.java

```diff
@@ -402,6 +403,39 @@ public QueryState getSynchronizedState() {
     }
   }
 
+  public boolean hasUnionPlan() throws Exception {
+    boolean result = false;
+
+    // During executing UNION statement or UNION ALL statement, the terminal block doesn't have join plan and has
+    // two more child blocks. In above case, Tajo should summarize result TableStats from all child stages.
+    ExecutionBlock terminalBlock = plan.getTerminalBlock();
+    terminalBlock.setPlan(terminalBlock.getPlan());
+    if (plan.getChilds(terminalBlock.getId()).size() >= 2) {
+      result = true;
+    }
+    return result;
+  }
+
+  public TableStats computeChildResultStats() throws Exception {
```

Comment:
Moving this method to ``StatisticsUtil`` throws compile error as following:
```
[ERROR] The projects in the reactor contain a cyclic reference: Edge between 'Vertex{label='org.apache.tajo:tajo-catalog-common:0.12.0-SNAPSHOT'}' and 'Vertex{label='org.apache.tajo:tajo-core:0.12.0-SNAPSHOT'}' introduces to cycle in the graph org.apache.tajo:tajo-core:0.12.0-SNAPSHOT --> org.apache.tajo:tajo-catalog-common:0.12.0-SNAPSHOT --> org.apache.tajo:tajo-core:0.12.0-SNAPSHOT -> [Help 1]
```

It seems that because there is a cyclic reference between tajo-catalog-common module and tajo-core module. So I improved this method using ``StatisticsUtil.aggregateTableStat`` instead of moving to ``StatisticsUtil``.